src/rai_bench/rai_bench/benchmark_model.py

```diff
@@ -134,10 +200,166 @@ def count_adjacent(self, positions: List[Pose], threshold_distance: float) -> in
 
         return adjacent_count
 
+    def build_neighbourhood_list(
+        self, entities: List[EntityT], threshold_distance: float = 0.15
+    ) -> Dict[EntityT, List[EntityT]]:
+        """
+        Build a neighbourhood list assigning a list of neighbours to every entity based on a threshold distance.
+
+        Parameters
+        ----------
+        entities : List[EntityT]
+            The list of entities.
+        threshold_distance : float, optional
+            The maximum distance between entities to consider them neighbours. Default is 0.15.
+
+        Returns
+        -------
+        Dict[EntityT, List[EntityT]]
+            A dictionary mapping each entity to a list of neighbouring entities.
+        """
+        neighbourhood_graph: Dict[EntityT, List[EntityT]] = {
+            entity: [] for entity in entities
+        }
+        for entity in entities:
+            neighbourhood_graph[entity] = [
+                other
+                for other in entities
+                if entity != other
+                and self.is_adjacent(entity.pose, other.pose, threshold_distance)
+            ]
+        return neighbourhood_graph
+
+    def group_entities_by_type(
+        self, entities: List[EntityT]
+    ) -> Dict[str, List[EntityT]]:
+        """
+        Group entities by their prefab type.
+
+        Parameters
+        ----------
+        entities : List[EntityT]
+            The list of entities to group.
+
+        Returns
+        -------
+        Dict[str, List[EntityT]]
+            A dictionary with keys as prefab names and values as lists of entities of that type.
+        """
+        entities_by_type: Dict[str, List[EntityT]] = {}
```

Comment:
I think it's more clear than `.setdefault`
```suggestion
        entities_by_type: Dict[str, List[EntityT]] = defaultdict(list)
```

Reply:
applied here:
https://github.com/RobotecAI/rai/pull/452/commits/57d8569506156d3f62ca3af698aadc5114be9708